app/Utils/DatabaseCleanupUtils.php

```diff
@@ -137,28 +140,12 @@ public static function removeBuild($buildid) : void
         ", array_merge($buildids, $buildids));
 
         // Delete the configure if not shared.
-        $build2configure = DB::select("
-                               SELECT a.configureid
-                               FROM build2configure AS a
-                               LEFT JOIN build2configure AS b ON (
-                                   a.configureid=b.configureid
-                                   AND b.buildid NOT IN $buildid_prepare_array
-                               )
-                               WHERE a.buildid IN $buildid_prepare_array
-                               GROUP BY a.configureid
-                               HAVING count(b.configureid)=0
-                           ", array_merge($buildids, $buildids));
-
-        $configureids = [];
-        foreach ($build2configure as $build2configure_array) {
-            // It is safe to delete this configure because it is only used
-            // by builds that are being deleted.
-            $configureids[] = intval($build2configure_array->configureid);
-        }
-        if (count($configureids) > 0) {
-            $configureids_prepare_array = $db->createPreparedArray(count($configureids));
-            DB::delete("DELETE FROM configure WHERE id IN $configureids_prepare_array", $configureids);
-        }
+        Configure::whereHas('builds', function (Builder $query) use ($buildids) {
+            $query->whereIn('id', $buildids);
+        })
+            ->whereDoesntHave('builds', function (Builder $query) use ($buildids) {
+                $query->whereNotIn('id', $buildids);
+            })->delete();
```

Comment:
We don't have any formatting rules to enforce this, but I personally think it's a bit cleaner to chain commands with anonymous functions like this:

```suggestion
        Configure::whereHas('builds', function (Builder $query) use ($buildids) {
            $query->whereIn('id', $buildids);
        })->whereDoesntHave('builds', function (Builder $query) use ($buildids) {
            $query->whereNotIn('id', $buildids);
        })->delete();
```

Reply:
agreed